routing/router.go

```diff
@@ -2729,11 +2729,16 @@ func (r *ChannelRouter) BuildRoute(amt *lnwire.MilliSatoshi,
 		return nil, err
 	}
 
-	// Allocate a list that will contain the unified policies for this
-	// route.
-	edges := make([]*unifiedPolicy, len(hops))
+	var (
```

Comment:
Instead of removing `newRoute` from `BuildRoute`, we can also skip that and re-route `BuildRoute` to pathfinding: https://github.com/lightningnetwork/lnd/pull/6921